src/biogeochem/FATESFireFactoryMod.F90

```diff
@@ -51,25 +52,57 @@ subroutine create_fates_fire_data_method( fates_fire_data_method )
 
-    current_case = fates_spitfire_mode
-
-    select case (current_case)
+    !
+    ! For FATES options that bypass fire...
+    !
+    if ( use_fates_sp .or. use_fates_ed_st3 )then
+      !
+      ! Make sure fire-emissions is NOT on
+      !
+      if ( shr_fire_emis_mechcomps_n > 0 )then
+         if ( use_fates_sp )then
+            write(iulog,*) "Fire emissions can NOT be on with FATES-SP mode: ", &
+                        errMsg(sourcefile, __LINE__)
+            call endrun(msg="Fire emission with FATES requires FATES to NOT be in Satellite Phenology (SP) mode" )
+         else if ( use_fates_ed_st3 )then
+            write(iulog,*) "Fire emissions can NOT be on with FATES ST3 mode: ", &
+                        errMsg(sourcefile, __LINE__)
+            call endrun(msg="Fire emission with FATES requires FATES to NOT be in Static Stand Structure mode" )
+         end if
+         ! For unit-testing return with a FATESFireData type, so there isn't a run-time error
+         ! Also do the FATESFireData type, as using FATESFireNoData type will fail with an error
+         allocate(fates_fire_data_type :: fates_fire_data_method)
+         return
+      end if
+      allocate(fates_fire_no_data_type :: fates_fire_data_method)
+    else
+      !
+      ! For regular FATES options that include fire
+      !
+      select case (fates_spitfire_mode)
 
-    case (no_fire:scalar_lightning)
-       allocate(fates_fire_no_data_type :: fates_fire_data_method)
-    case (lightning_from_data:anthro_suppression)
-       allocate(fates_fire_data_type :: fates_fire_data_method)
+         ! No-fire, scalar-lightning and successful_ignitions ALL do NOT need input data from the base class
+         case (no_fire:scalar_lightning)
+            allocate(fates_fire_no_data_type :: fates_fire_data_method)
+         case (successful_ignitions)
+            allocate(fates_fire_no_data_type :: fates_fire_data_method)
+         ! Lightning from data, and the anthro types (ignition and suppression) need lightning data from the base class
+         case (lightning_from_data)
+            allocate(fates_fire_data_type :: fates_fire_data_method)
+         case (anthro_ignitions:anthro_suppression)
+            allocate(fates_fire_data_type :: fates_fire_data_method)
 
-    case default
-       write(iulog,*) subname//' ERROR: unknown method: ', fates_spitfire_mode
-       call endrun(msg=errMsg(sourcefile, __LINE__))
+         case default
+            write(iulog,*) 'Unrecognized fates_spitfire_mode option = ', fates_spitfire_mode, ' in: ', &
+                           errMsg(sourcefile, __LINE__)
+            call endrun(msg="Unknown option for namelist item fates_spitfire_mode:")
+            ! For unit-testing, make sure a valid fates_fire_data_method is set and return, otherwise it fails with a seg-fault
+            allocate(fates_fire_no_data_type :: fates_fire_data_method)
 
-    end select
+      end select
+    end if
```

Comment:
For now we also need to add a check for FATES turned on at all, because until we bring in changes it can't handle fire-emis with FATES.